[ADP-3368] Move E2E tests in main pipeline

paolino opened the pull request:
- [x] Move E2E tests for linux and macos from `release` to the main `pipeline`
- [x] Move all pipelines scripts into the scripts folder
- [x] Move the docker steps into the main pipeline
- [x] Remove `tagly` and `nightly`cpipelines
- [x] Fix blocks in the main pipeline to not block on release candidate runs
- [x] Add a trigger from the release pipeline to the main pipeline
- [x] Add concurrency control to the steps in the main pipeline
- [x] Trigger the windows GHA from buildkite once the windows artifacts are ready 
- [x] Add concurrency control to the windows E2E GHA

ADP-3368

Commit:
Integrate all release steps to main buildkite pipeline

Commit:
Apply shellcheck lints

Commit:
Clean up release pipeline

Commit:
Use release-candidate name when releasing from master and test-rc when otherwise

Commit:
Move push docker image to main pipeline

Commit:
Delete nightly pipeline

Commit:
Clean up blocks in main pipeline

Commit:
Limit jobs concurrency on critical steps

Commit:
Add triggering to release pipeline

Commit:
Fix restoration benchmark

Commit:
Change windows E2E GHA to react on release tags

Commit:
Change windows unit test GHA to reaact on release tags

Commit:
Move windows trigger tag creation in the windows step group

HeinrichApfelmus commented on scripts/buildkite/main/tag-release-candidate.sh:
```diff
@@ -0,0 +1,12 @@
+git remote set-url origin \
+  "https://$GITHUB_TOKEN@github.com/cardano-foundation/cardano-wallet.git"
+git tag "$DTAG" "$BUILDKITE_COMMIT"
+git push origin "$DTAG"
```
```suggestion
git push origin "$DTAG"

```

(newline at end of file)

HeinrichApfelmus commented on .buildkite/pipeline.yml:
```diff
@@ -4,375 +4,437 @@ agents:
+      agents:
+        system: ${linux}
+
+  - group: "Linux Tests"
```
```suggestion
  - group: "Linux Checks"
```

Could you adjust to the nomenclature from "Continuous Integration :: AP"?

HeinrichApfelmus commented on .buildkite/pipeline.yml:
```diff
@@ -4,375 +4,437 @@ agents:
-      TMPDIR: "/cache"
-
-  - label: Manage rc-latest tags
+  - group: "Linux Setup"
```
```suggestion
  - group: "Linux Artifacts"
```

Could you adjust to the nomenclature of "Continuous Integration :: AP"? (Same for the other groups.)

Commit:
Fix steps naming in main pipeline

Commit:
Block integration tests on macos

HeinrichApfelmus committed:
Amend Restore benchmark (linux)

HeinrichApfelmus commented on scripts/buildkite/release/generate-trigger.sh:
```diff
@@ -0,0 +1,28 @@
+        message: Release Candidate of $VERSION
+        env:
+            RELEASE_CANDIDATE: "$VERSION"
+            BUILDKITE_BRANCH: release-candidate-$VERSION
```
```suggestion
            BUILDKITE_BRANCH: release-candidate/$VERSION
```

? 🤔 

I have the feeling that setting `BUILDKITE_BRANCH` to a name that is not actually a branch name will lead to trouble later.

HeinrichApfelmus commented on .buildkite/pipeline.yml:
```diff
@@ -4,375 +4,435 @@ agents:
+      command: 'nix build -L .#ci.aarch64-darwin.tests.run.unit'
+      agents:
+        system: ${macos}
+
```
To be consistent with the nomenclature, I recommend moving the unit tests to the "MacOS Checks" section.

Commit:
Move macos unit tests into macos checks

paolino committed:
[ADP-3368] Amend Restore benchmark (linux) (#4664)

This pull request amends the "Amend Restore benchmark (linux)" step to
work correctly.

### Issue Number

ADP-3368
